internal/cloud/backend_plan.go

```diff
@@ -324,9 +324,20 @@ in order to capture the filesystem context the remote workspace expects:
 	}
 
 	// Retrieve the run to get its current status.
-	r, err = b.client.Runs.Read(stopCtx, r.ID)
+	runID := r.ID
+	r, err = b.client.Runs.ReadWithOptions(stopCtx, runID, &tfe.RunReadOptions{
+		Include: "task_stages",
+	})
 	if err != nil {
-		return r, generalError("Failed to retrieve run", err)
+		// This error would be expected for older versions of TFE that do not allow
+		// fetching task_stages.
+		if strings.HasSuffix(err.Error(), "Invalid include parameter") {
+			r, err = b.client.Runs.Read(stopCtx, runID)
+		}
```

Comment:
This feels brittle to me. Although it's not ideal, the 'proper' way to do this would be to use the `TFP-API-Version` header information (exposed in the client [here](https://github.com/hashicorp/go-tfe/blob/73c3b410274fc2d42eeb60ded9157ae325484301/tfe.go#L280-L298)) to conditionally make a single request if the host (TFC or TFE) supports the necessary API version for it.

Reply:
That is my intention! Once this is merged I can supplant the next version with run tasks. Feels bad though because I know run tasks will squat on that version for a long time. https://github.com/hashicorp/atlas/pull/11572

Reply:
If we hit line 335, and TFE or TFC have Run Tasks defined for that workspace, then the CLI would not display anything about run tasks' results but the platform would still show run tasks' results, which is the discrepancy/problem we were trying to solve to begin with. I wasn't sure if that's the behavior we want, but I want to bring that up just in case. I understand these changes are temporary. @brandonc @chrisarcand

Reply:
If we hit line 335, run tasks won't be defined for the workspace because we'll be working against a version of TFE that doesn't support them -- that's what "invalid include parameter" would indicate... But it's not a great feature detection strategy and I'd like to rely on the API version instead.